src/core/src/op/select.cpp

```diff
@@ -41,10 +39,9 @@ void op::v1::Select::validate_and_infer_types() {
                           element::Type::merge(result_et, get_input_element_type(1), get_input_element_type(2)),
                           "Argument 1 and 2 element types must match.");
 
-    std::vector<ov::PartialShape> output_shapes = {ov::PartialShape{}};
-    const std::vector<ov::PartialShape> input_shapes = {get_input_partial_shape(0),
-                                                        get_input_partial_shape(1),
-                                                        get_input_partial_shape(2)};
+    const auto input_shapes = get_node_input_partial_shapes(*this);
+    std::vector<ov::PartialShape> output_shapes{ov::PartialShape::dynamic()};
```

Comment:
```suggestion
    auto output_shapes = std::vector<ov::PartialShape>(1);
```
to use `auto`. Can also add optionally 2nd parameter (`PartialShape{}`) of vector ctor.
It doesn't matter what is inside the vector it must have required number of elements only, `shape_infer` will set correct shape if success. So default initialization is simplest.

Reply:
Done